File: Objective-C/Dark Tower/Classes/Random.m
//
//  Random.m
//  DarkTower
//
//  Created by Louis Revor on 2/19/10.
//  Copyright 2010 Louis Revor. All rights reserved.
//

#import "Random.h"
#import <stdlib.h>
#import <time.h>

static BOOL initialized = FALSE;

@implementation Random
+ (void) seed {
	time_t myTime;
	time(&myTime);
	srand(myTime);
	initialized=TRUE;
}

+ (NSInteger) getRand {
	return [Random getRand:10];
}

+ (NSInteger) getRand:(NSInteger) maxVal {
	float floatVal;
	NSInteger value;
	if (!initialized) [Random seed];
	floatVal = rand();
	floatVal=floatVal/RAND_MAX;
	value=(NSInteger) (floatVal*maxVal+1);
	return value;
}
@end

File: Swift/DarkTowerSwift/DarkTower/Random.swift
//
//  Random.swift
//  DarkTower
//
//  Created by Louis Revor on 3/1/15.
//  Copyright (c) 2015 Louis Revor. All rights reserved.
//

import Foundation

class Random {
    func getRand() -> NSInteger {
        NSLog("[%@ %@]", reflect(self).summary, __FUNCTION__)
        return(getRand(10))
    }

    func getRand(maxVal:NSInteger) ->NSInteger {
        NSLog("[%@ %@]", reflect(self).summary, __FUNCTION__)

        var maxInt:UInt32 = UInt32(maxVal)
        var random = arc4random_uniform(maxInt)
        var value:NSInteger = NSInteger(random) + 1
        NSLog("[%@ %@] Range=%d, RandomValue=%d", reflect(self).summary, __FUNCTION__, maxVal, value)
        return(value)
    }
    
}
